source/common/network/lc_trie.h

```diff
@@ -99,11 +99,23 @@ class LcTrie {
   typedef uint32_t Ipv4;
   typedef absl::uint128 Ipv6;
 
+  typedef std::unordered_set<std::string> TagSet;
+  typedef std::shared_ptr<std::unordered_set<std::string>> TagSetSharedPtr;
+
   /**
    * Structure to hold a CIDR range and the tag associated with it.
    */
   template <class IpType, uint32_t address_size = CHAR_BIT * sizeof(IpType)> struct IpPrefix {
 
+    IpPrefix() {}
```

Comment:
I think a default constructor is dangerous unless we also have a default initializor for `length_`.